### PR TITLE
Fix #977

### LINKED ISSUE #977: [Bug] [app-reg-tool] - In blazorwasm apps, Client/Program.cs "api://api.id.uri/access_as_user" is incorrectly read/writen

### PR DESCRIPTION
**Pull request by jmprieur**
Fixes #977

### CODE REVIEW

File: tools/app-provisioning-tool/app-provisioning-lib/MicrosoftIdentityPlatformApplication/MicrosoftIdentityPlatformApplicationManager.cs [hunk 1]
```diff
@@ -185,6 +186,7 @@ internal async Task UpdateApplication(TokenCredential tokenCredential, Applicati
                 .UpdateAsync(updatedApp).ConfigureAwait(false);
 
             if (existingApplication.RequiredResourceAccess != null
+                && !reconcialedApplicationParameters.IsBlazorWasm
```
Comment (jmprieur) [inline tools/app-provisioning-tool/app-provisioning-lib/MicrosoftIdentityPlatformApplication/MicrosoftIdentityPlatformApplicationManager.cs:188]: In Blazorwasm hosted, the SPA does not need a password.

**Review by jennyf19 — APPROVED** [review]

File: tools/app-provisioning-tool/app-provisioning-lib/MicrosoftIdentityPlatformApplication/MicrosoftIdentityPlatformApplicationManager.cs [hunk 2]
```diff
@@ -380,13 +382,19 @@ await graphServiceClient.Oauth2PermissionGrants
         /// <param name="application"></param>
         private static void AddSpaPlatform(ApplicationParameters applicationParameters, Application application)
         {
+#if GRAPH_SDK_SUPPORTS_SPA
+            application.Spa = new SpaApplication();
```
Comment (jmprieur) [inline tools/app-provisioning-tool/app-provisioning-lib/MicrosoftIdentityPlatformApplication/MicrosoftIdentityPlatformApplicationManager.cs:384]: The Graph SDK now has support for SPA

File: tools/app-provisioning-tool/app-provisioning-lib/MicrosoftIdentityPlatformApplication/MicrosoftIdentityPlatformApplicationManager.cs [hunk 3]
```diff
@@ -1,6 +1,7 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
+#define GRAPH_SDK_SUPPORTS_SPA
```
Comment (jennyf19) [inline tools/app-provisioning-tool/app-provisioning-lib/MicrosoftIdentityPlatformApplication/MicrosoftIdentityPlatformApplicationManager.cs:4]: remove?

File: tools/app-provisioning-tool/app-provisioning-lib/ProjectDescriptions/dotnet-blazorwasm-hosted.json [hunk 4]
```diff
@@ -3,225 +3,6 @@
   "ProjectRelativeFolder": "",
   "BasedOnProjectDescription": "",
   "ConfigurationProperties": [
```
Comment (jmprieur) [inline tools/app-provisioning-tool/app-provisioning-lib/ProjectDescriptions/dotnet-blazorwasm-hosted.json:5]: No need for the toplevel blazorwasm hosted to get properties. They are delegated to the sub folders

File: tools/app-provisioning-tool/app-provisioning-lib/Tool/AppProvisioningTool.cs [hunk 5]
```diff
@@ -40,73 +40,66 @@ public async Task Run()
...
                 projectDescription,
                 ProjectDescriptionReader.projectDescriptions);
 
-            if (!projectSettings.ApplicationParameters.IsB2C && !string.IsNullOrEmpty(ProvisioningToolOptions.SusiPolicyId))
+            // Case of a blazorwasm hosted application. We need to create two applications:
+            // - the hosted web API
+            // - the SPA.
+            if (projectSettings.ApplicationParameters.IsBlazorWasm && projectSettings.ApplicationParameters.IsWebApi)
             {
-                // Get all the files in which "AzureAD" needs to be replaced by "AzureADB2C"
```
Comment (jmprieur) [inline tools/app-provisioning-tool/app-provisioning-lib/Tool/AppProvisioningTool.cs:58]: Extracted as a method below

**Pull request merged**